simpledotorg/simple-server
# Maharashtra DHIS2 gender support

**Story card:** [sc-5894](https://app.shortcut.com/simpledotorg/story/5894/add-gender-disaggregation-to-the-dhis2-data-elements-we-report)

## Because

We need to report per-gender indicators for monthly registrations and monthly controlled patients to Maharashtra's DHIS2 test server. This is to conform to the existing NPCDCS indicators that we will be feeding into, along with the CPHC app and other sources.

## This addresses

* Introduction of a one-off service called the MaharashtraDHIS2Exporter.
* Addition of two gender-specific methods into the Reports::Repository
* Wiring it all up to shoot these two indicators (regns and controlled patients, male and female) to the DHIS2 test server.

The PR introduces sort of "hacky" approaches with the philosophy that the duplication of logic is worth it, so we can easily rip this stuff out if needed. We may take a full-fledged approach that allows the reporting pipeline to support gender-specific indicators in the future, but that's a much larger effort.

This PR just gets data into DHIS2 so that we can get that quick win with stakeholders and unlock the WHO to start advocating for other data sources to send their indicators to DHIS2 as well.

## Test instructions

* Grab MH DHIS2 credentials locally, send a local facility's indicators to the Maharashtra DHIS2 server, verify that they reach in one piece.
* Ship the PR, let prod data get shipped to DHIS2 (overwriting the test data we sent earlier), verify that prod data also makes it to DHIS2 successfully
